dependency_support/com_github_quantamhd_lemon/lemon.patch

```diff
@@ -0,0 +1,2945 @@
++#include "lemon/christofides_tsp.h"
++#include "lemon/opt2_tsp.h"
+ 
+ #include "test_tools.h"
+ 
+diff --git a/test/unionfind_test.cc b/test/unionfind_test.cc
+index e82d8e6..08a8285 100644
+--- a/test/unionfind_test.cc
++++ b/test/unionfind_test.cc
+@@ -16,9 +16,9 @@
+  *
+  */
+ 
+-#include <lemon/list_graph.h>
+-#include <lemon/maps.h>
+-#include <lemon/unionfind.h>
++#include "lemon/list_graph.h"
++#include "lemon/maps.h"
++#include "lemon/unionfind.h"
+ #include "test_tools.h"
+ 
+ using namespace lemon;
+diff --git a/tools/dimacs-solver.cc b/tools/dimacs-solver.cc
+index 60da233..e0bef5f 100644
+--- a/tools/dimacs-solver.cc
++++ b/tools/dimacs-solver.cc
+@@ -32,18 +32,18 @@
+ #include <fstream>
+ #include <cstring>
+ 
+-#include <lemon/smart_graph.h>
+-#include <lemon/dimacs.h>
+-#include <lemon/lgf_writer.h>
+-#include <lemon/time_measure.h>
++#include "lemon/smart_graph.h"
++#include "lemon/dimacs.h"
++#include "lemon/lgf_writer.h"
++#include "lemon/time_measure.h"
+ 
+-#include <lemon/arg_parser.h>
+-#include <lemon/error.h>
++#include "lemon/arg_parser.h"
++#include "lemon/error.h"
+ 
+-#include <lemon/dijkstra.h>
+-#include <lemon/preflow.h>
+-#include <lemon/matching.h>
+-#include <lemon/network_simplex.h>
++#include "lemon/dijkstra.h"
++#include "lemon/preflow.h"
++#include "lemon/matching.h"
++#include "lemon/network_simplex.h"
+ 
+ using namespace lemon;
+ typedef SmartDigraph Digraph;
+diff --git a/tools/dimacs-to-lgf.cc b/tools/dimacs-to-lgf.cc
+index 3968645..cf0986f 100644
+--- a/tools/dimacs-to-lgf.cc
++++ b/tools/dimacs-to-lgf.cc
+@@ -33,12 +33,12 @@
+ #include <fstream>
+ #include <cstring>
+ 
+-#include <lemon/smart_graph.h>
+-#include <lemon/dimacs.h>
+-#include <lemon/lgf_writer.h>
++#include "lemon/smart_graph.h"
++#include "lemon/dimacs.h"
++#include "lemon/lgf_writer.h"
+ 
+-#include <lemon/arg_parser.h>
+-#include <lemon/error.h>
++#include "lemon/arg_parser.h"
++#include "lemon/error.h"
+ 
+ using namespace std;
+ using namespace lemon;
+diff --git a/tools/lgf-gen.cc b/tools/lgf-gen.cc
+index 390c8ae..07d11cc 100644
+--- a/tools/lgf-gen.cc
++++ b/tools/lgf-gen.cc
+@@ -31,19 +31,19 @@
+ #include <algorithm>
+ #include <set>
+ #include <ctime>
+-#include <lemon/list_graph.h>
+-#include <lemon/random.h>
+-#include <lemon/dim2.h>
+-#include <lemon/bfs.h>
+-#include <lemon/counter.h>
+-#include <lemon/suurballe.h>
+-#include <lemon/graph_to_eps.h>
+-#include <lemon/lgf_writer.h>
+-#include <lemon/arg_parser.h>
+-#include <lemon/euler.h>
+-#include <lemon/math.h>
+-#include <lemon/kruskal.h>
+-#include <lemon/time_measure.h>
++#include "lemon/list_graph.h"
++#include "lemon/random.h"
++#include "lemon/dim2.h"
++#include "lemon/bfs.h"
++#include "lemon/counter.h"
++#include "lemon/suurballe.h"
++#include "lemon/graph_to_eps.h"
++#include "lemon/lgf_writer.h"
++#include "lemon/arg_parser.h"
++#include "lemon/euler.h"
++#include "lemon/math.h"
++#include "lemon/kruskal.h"
++#include "lemon/time_measure.h"
+ 
```

Comment:
Sorry @QuantamHD about the slow review. This kind of issue I have otherwise resolved by using either the `includes` attribute of `cc_library`, or if that doesn't work `copts`. No need to change that now, but a patch like this is probably going to be a bit of a pain to maintain with version bumps.

Reply:
So the issue is that bazel doesn't allow you to `includes` angle includes. The copts solution would work, but it has the disadvantage that all future importers have to also include those copts in their cc_* rules. Also this library hasn't been updated in over 7 years (2014-07-07) so I'm not too concerned about the patch size.

Reply:
I misremembered: It is `include_prefix` and `strip_include_prefix` that you can (hackily) use to work around this. If they are used it's possible to import with `<>`.

Reply:
But it seems fine to leave this as is :smile: